### DISCUSSION TITLE
Custom prefix autocomplete

### DISCUSSION THREAD

**Discussion by royassis** [apache/jena#2345] [Q&A]
Hey

I see that when writing the word **PREFIX** in the UI's SPARQL windows there a pop up with prefixes autocomplete:
![image](https://github.com/apache/jena/assets/34189329/1a8c0033-e57e-4caa-92d5-18105cb9de47)

Is it possible to add a custom prefix?

**Comment by kinow** [comment]
I **think** YASGUI remembers the one you use, but I could be wrong, or maybe I saw it in another library or component.

But if you want to provide a list of prefixes, then that feature is still missing. We had a similar question some time ago, https://github.com/apache/jena/discussions/1680, to which @SimonBin created a PR https://github.com/apache/jena/pull/1683

Feel free to comment in the pull request if you can test it or if you have suggestions to help we move forward with it (I do not recall much about it right now).

**Comment by SimonBin** [comment]
if I remember correctly, all you need to do is register your prefix on prefix.cc and clear your local storage/browser cache
